batch/batch/worker/worker.py

```diff
@@ -1088,36 +1063,50 @@ def get_next_xfsquota_project_id():
         Job.quota_project_id += 1
         return project_id
 
-    def secret_host_path(self, secret):
+    def secret_host_path(self, secret) -> str:
         return f'{self.scratch}/secrets/{secret["name"]}'
 
-    def io_host_path(self):
+    def io_host_path(self) -> str:
         return f'{self.scratch}/io'
 
-    def gcsfuse_path(self, bucket):
+    def cloudfuse_base_path(self):
+        # Make sure this path isn't in self.scratch to avoid accidental bucket deletions!
+        return f'/cloudfuse/{self.token}'
+
+    def cloudfuse_data_path(self, location: str) -> str:
+        # Make sure this path isn't in self.scratch to avoid accidental bucket deletions!
```

Comment:
The real point here is that we can't `rm -rf` this path. It's not trivial to assert that one path does not resolve to a folder that is "inside" another because of hard links and symlinks.

While I sympathize with the need for a good assertion, I'm not sure how to do that here.